### DISCUSSION TITLE
It's frustrating, to click a "view more" button, to reveal valuable content

### DISCUSSION THREAD

**Discussion by pmario** [community/community#3286] [Discussions]
If there is a "brainstorming" discussion, with a lot of back and forth, the viewer adds several **view more** buttons. 

It's super frustrating that I have to use the "browser search" to find and expand them 1 by 1, if I want to have a look at the ideas, that have come up in the discussion. 

**It would be nice to have a global "expand all" button**

**Comment by evi-liu** [comment]
Thanks for the feedback! We hear you, and we're currently working on improving the threading design including the "view more" functionality.

**Reply by pmario** [reply]
Thank you for the good work! and the fast reply!

**Reply by cambragol** [reply]
I think this could be improved as well. I didn't even notice the 'read more' text, and was frequently wondering where new comments I had received emails for, were.

Perhaps a 'collapse/expand' button, that is open by default. Your current implementation assumes the user does not want to see certain comments, which is something the user should get to decide. The function of the 'discussion' is to display discussions first. Trimming those down or modifying them in some way should be secondary.

**Comment by aschrijver** [comment]
Yes, I agree. The "view more" is antithetical to a good discussion forum, imho. Maybe if some devs like things to be short, it may be a Settings thing to enable it, but otherwise I personally would love to see entire discussion threads. @cambragol expand/collapse is a good idea, say similar to how Hacker News does it (remembers collapsed branches after a reload).

**Comment by actual-nh** [comment]
I have also noticed that, if I start a comment that will be in the "view more" section, and check another website for something, then come back, my draft comment will not be present (including after clicking "view more"). Normally, I can start a comment, go to another website in that tab, come back, and find the draft still there (saved by the browser, I suspect - Chrome on Mac).

**Comment by aoterolorenzo** [comment]
Noticed that also. I think would be more interesting to have a "view more" button that would shown the oldest comments between the first and the rest of the comments, showing only the newest ones.

**Comment by Kochise** [comment]
Problem is, even for short discussion you have that annoying "View more" clickable text :

https://github.com/hoffstadt/DearPyGui/discussions/935

What's the point ? Show the whole damn discussion : it's Github, not Reddit, Tweeter or Disqus.

**Reply by pmario** [reply]
It may make a difference, if you need to service 10 requests an hour or 1 million ... may be ;) .... BUT I would be OK if there would be a "top level" "view all" button, that loads the rest if I need it.

**Reply by verdverm** [reply]
I second having the default be to have all comments shown, and have a "hide thread" button instead

- This is how issues work
- the "view more" is tiny and easy to miss
- the most recent replies are the ones hidden, so it always needs to be clicked once a thread is sufficiently (barely) "long"

**Reply by pmario** [reply]
The problem is, that the thread hasn't to be long. ... IMO it starts cutting with 2 or 3 replies :/

**Reply by verdverm** [reply]
indeed, this is why I put "long" in quotation marks, and hopefully this thread becomes long enough to trigger the effect :]

**Reply by pmario** [reply]
May be that's the reason they didn't fix it ;) ... It doesn't happen to them q:-)

**Reply by verdverm** [reply]
Third hidden comment, @evi-liu do you see how this is a subpar UX?

**Comment by Jay-o-Way** [comment]
Is this still an issue?

**Reply by pmario** [reply]
I think it somewhat improved, since now the "5 latest replies" are shown and the "older" ones are hidden. .. Scroll up a bit and see the "hidden replies" above this comment.

**The point** in the OP was that **"hidden comments" contain valuable information**. 

I want to be able to use discussions for public technical brainstorming. So there can be a lot of back and forth with 200++ comments and replies. 

I usually use the browser built in search function with CTRL-F because I do need the browser "search highlight" function ... Search results should be visible, but _not filtered_ to see the whole context.

I have to do the following: 

- CTRL-F
- Search for "**hidden items**" to expand them
![image](https://user-images.githubusercontent.com/374655/190864374-f22c20ad-09e9-4c51-b6ea-f25047a38ee1.png)
- Search for "**previous reply**" **several times** until I could open them all
- Make sure I found them all ... 
- **Finally** ... Search for the "term" I'm interested in. 
- ... If it's not there go to the next thread and start at step 1

>Is this still an issue?

So from my point of view ... **yea -- it still is an issue**.

**Reply by verdverm** [reply]
**Absolutely still an issue**, you just swapped where the problem is by moving it from the last ones to the first ones. Did you consider that often the first comment is the most important comment _and_ the last comment can also be the most important comment.

**Don't trim them at all and let the repository admins hide any comments just like Issues. Simpler and more consistent.** While you're at it, it would be great if discussions could also be started like a PR, so we could discuss a code change or diff without opening a PR, triggering actions, etc.
